api/src/services/connected-user-log.js

```diff
@@ -0,0 +1,37 @@
+const db = require('../db');
+
+const LOG_TYPE = 'connected-user-';
+const UPDATE_TIME_INTERVAL = 30 * 60 * 1000; // 30 minutes
+
+const saveLog = (user) => {
+  if (!user) {
+    const error = new Error('Error when saving log: Log Information missing');
+    return Promise.reject(error);
+  }
+  
+  const id = LOG_TYPE + user;
+  const log = {
+    user,
+    timestamp: new Date().getTime()
+  };
+
+  return db.medicLogs
+    .get(id)
+    .catch(error => {
+      if (error.status === 404) {
+        return { _id: id };
+      }
+      throw error;
+    })
+    .then(doc => {
+      if (log.timestamp - doc.timestamp < UPDATE_TIME_INTERVAL) {
```

Comment:
With new docs, `doc.timestamp` will be undefined, which means `log.timestamp - doc.timestamp` is `NaN` which means the whole expression does evaluate to `false` so everything works. But it wasn't clear to me that this was going to work just from reading it, I had to try it out in a repl! I think it would be more obvious if it was like...

```suggestion
      if (doc.timestamp && log.timestamp - doc.timestamp < UPDATE_TIME_INTERVAL) {
```